app/src/main/java/com/lagradost/cloudstream3/ui/home/HomeFragment.kt

```diff
@@ -590,7 +592,16 @@ class HomeFragment : Fragment() {
 
                     is Resource.Failure -> {
                         homeLoadingShimmer.stopShimmer()
-                        resultErrorText.text = data.errorString
+                        var errorString =  data.errorString
+                        if (context?.isNetworkAvailable() == false) {
```

Comment:
Just use isNetworkError or extend the Resource.Failure and check the exception type of the error. This is because isNetworkAvailable is not always correct.

Reply:
That actually seems to be more inaccurate though. IE if DNS fails but you actually have connection it is the same error as it is a network error, whereas what I currently use does it only if device has no service.